internal/api/v1/admin/accesskeys/access_keys_api.go

```diff
@@ -0,0 +1,61 @@
+package accesskeys
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/errutil"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/querybuilders"
+	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
+	"github.com/go-resty/resty/v2"
+)
+
+const route = "/api/v1/admin/users/keys"
+
+type API struct {
+	httpClient *resty.Client
+	url        *url.URL
+}
+
+func (a *API) AccessKeys(ctx context.Context, opts ...queries.AdminAccessKeyQueryOption) (*queries.AccessKeyPage, error) {
+	var query queries.AdminAccessKeyQuery
+	for _, o := range opts {
+		o(&query)
+	}
+
+	queryBuilder := querybuilders.NewQueryBuilder(
+		querybuilders.WithMetadataFilter(query.Metadata),
+		querybuilders.WithPageFilter(query.PageFilter),
+		querybuilders.WithFilterQueryBuilder(&adminAccessKeyFilterQueryBuilder{adminAccessKeyFilter: query.AdminAccessKeyFilter}),
+	)
+	params, err := queryBuilder.Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build access keys query params: %w", err)
+	}
+
+	var result queries.AccessKeyPage
+	_, err = a.httpClient.
+		R().
+		SetContext(ctx).
+		SetResult(&result).
+		SetQueryParams(params.ParseToMap()).
+		Get(a.url.String())
+	if err != nil {
+		return nil, fmt.Errorf("HTTP response failure: %w", err)
+	}
+
+	return &result, nil
+}
+
+func NewAPI(url *url.URL, httpClient *resty.Client) *API {
```

Comment:
```suggestion
func NewAPI(url *url.URL, httpClient *resty.Client) *API {
if baseURL == nil {
		return nil, fmt.Errorf("url cannot be nil")
	}
	if httpClient == nil {
		return nil, fmt.Errorf("httpClient cannot be nil")
	}
```
just adding checks per nil